config/main.py

```diff
@@ -7031,5 +7032,68 @@ def del_subinterface(ctx, subinterface_name):
     except JsonPatchConflict as e:
         ctx.fail("{} is invalid vlan subinterface. Error: {}".format(subinterface_name, e))
 
+
+#
+# 'clock' group ('config clock ...')
+#
+@config.group()
+def clock():
+    """Configuring system clock"""
+    pass
+
+
+def get_tzs(ctx, args, incomplete):
+    ret = clicommon.run_command('timedatectl list-timezones',
+                                display_cmd=False, ignore_error=False,
+                                return_cmd=True)
+    if len(ret) == 0:
+        return []
+
+    lst = ret[0].split('\n')
+    if 'Etc/UTC' not in lst:
+        lst.append('Etc/UTC')
```

Comment:
This append doesn't make sense to me.

If Etc/UTC is not on the supported list, then it shouldn't be accepted as input. Otherwise, you don't need to append it.

Reply:
`Etc/UTC` is valid timezone.
Here you can see an example which shows that you can set both `UTC` and `Etc/UTC`:
```bash
admin@sonic:~$ sudo timedatectl set-timezone Europe/Kiev
admin@sonic:~$ timedatectl show
Timezone=Europe/Kiev
LocalRTC=no
CanNTP=no
NTP=no
NTPSynchronized=no
TimeUSec=Mon 2023-05-22 14:12:46 EEST
RTCTimeUSec=Mon 2023-05-22 14:12:40 EEST
```
```bash
admin@sonic:~$ timedatectl list-timezones | grep UTC
UTC
admin@sonic:~$ sudo timedatectl set-timezone ETC
Failed to set time zone: Invalid or not installed time zone 'ETC'
admin@sonic:~$ sudo timedatectl set-timezone UTC
admin@sonic:~$ timedatectl show
Timezone=UTC
LocalRTC=no
CanNTP=no
NTP=no
NTPSynchronized=no
TimeUSec=Mon 2023-05-22 11:13:22 UTC
RTCTimeUSec=Mon 2023-05-22 11:13:16 UTC
```
```bash
admin@sonic:~$ sudo timedatectl set-timezone Etc/UTC
admin@sonic:~$ timedatectl show
Timezone=Etc/UTC
LocalRTC=no
CanNTP=no
NTP=no
NTPSynchronized=no
TimeUSec=Mon 2023-05-22 11:13:29 UTC
RTCTimeUSec=Mon 2023-05-22 11:13:23 UTC
```
```bash
admin@sonic:~$ sudo timedatectl set-timezone ETC/UTC
Failed to set time zone: Invalid or not installed time zone 'ETC/UTC'
admin@sonic:~$
```

Reply:
Can you upgrade the timedatectrl to a version that will show Etc/UTC in the list?

```
yinxi@yinxi-vm0:~/src$ timedatectl list-timezones | grep UTC
Etc/UTC
UTC
yinxi@yinxi-vm0:~/src$ timedatectl --version
systemd 245 (245.4-4ubuntu3.21)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD +IDN2 -IDN +PCRE2 default-hierarchy=hybrid
```

Reply:
The `timedatectl` is not a problem here. The timezone information is provided only with `tzdata` package. And we are using the latest one. 
```
admin@r-anaconda-24:~$ sudo apt show tzdata
Package: tzdata
Version: 2021a-1+deb11u10
Priority: required
Section: localization
Maintainer: GNU Libc Maintainers <debian-glibc@lists.debian.org>
Installed-Size: 3,495 kB
Provides: tzdata-bullseye
Depends: debconf (>= 0.5) | debconf-2.0
Homepage: https://www.iana.org/time-zones
Tag: role::app-data, use::timekeeping
Download-Size: 286 kB
APT-Manual-Installed: yes
APT-Sources: http://debian-archive.trafficmanager.net/debian bullseye/main amd64 Packages
Description: time zone and daylight-saving time data

N: There are 9 additional records. Please use the '-a' switch to see them.
```

Moreover, if you will list all the timezone records (provided by `tzdata`) on the filesystem you will see that `UTC` is just a symlink to `Etc/UTC`:
```
admin@r-anaconda-24:~$ ls -la /usr/share/zoneinfo/
total 196
~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~
-rw-r--r--  1 root root    118 Apr 18 20:03 EST
-rw-r--r--  1 root root   2294 Apr 18 20:03 EST5EDT
drwxr-xr-x  2 root root    478 May 23 23:25 Etc
~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~ ~
drwxr-xr-x  2 root root    206 May 23 23:25 US
lrwxrwxrwx  1 root root      7 Apr 18 20:03 UTC -> Etc/UTC
-rw-r--r--  1 root root   1873 Apr 18 20:03 WET
lrwxrwxrwx  1 root root     13 Apr 18 20:03 W-SU -> Europe/Moscow
-rw-r--r--  1 root root  17889 Apr 18 20:03 zone1970.tab
-rw-r--r--  1 root root  19375 Apr 18 20:03 zone.tab
lrwxrwxrwx  1 root root      7 Apr 18 20:03 Zulu -> Etc/UTC
```

And even more, when I try to reconfigure `tzdata` I got the next output:
```
admin@r-anaconda-24:~$ sudo dpkg-reconfigure tzdata
debconf: unable to initialize frontend: Dialog
debconf: (No usable dialog-like program is installed, so the dialog based frontend cannot be used. at /usr/share/perl5/Debconf/FrontEnd/Dialog.pm line 78.)
debconf: falling back to frontend: Readline
Configuring tzdata
------------------

Please select the geographic area in which you live. Subsequent configuration questions will narrow this down by presenting a list of cities, representing the time zones in which they are located.

  1. Africa  2. America  3. Antarctica  4. Australia  5. Arctic Ocean  6. Asia  7. Atlantic Ocean  8. Europe  9. Indian Ocean  10. Pacific Ocean  11. US  12. None of the above
Geographic area: 12

Please select the city or region corresponding to your time zone.

  1. GMT    4. GMT+10  7. GMT+2  10. GMT+5  13. GMT+8  16. GMT-1   19. GMT-12  22. GMT-2  25. GMT-5  28. GMT-8  31. Greenwich  34. Universal
  2. GMT+0  5. GMT+11  8. GMT+3  11. GMT+6  14. GMT+9  17. GMT-10  20. GMT-13  23. GMT-3  26. GMT-6  29. GMT-9  32. UCT        35. Zulu
  3. GMT+1  6. GMT+12  9. GMT+4  12. GMT+7  15. GMT-0  18. GMT-11  21. GMT-14  24. GMT-4  27. GMT-7  30. GMT0   33. UTC
Time zone: 33


Current default time zone: 'Etc/UTC'
Local time is now:      Fri May 26 12:37:38 UTC 2023.
Universal Time is now:  Fri May 26 12:37:38 UTC 2023.

```
`Current default time zone: 'Etc/UTC'`

So I don't think we need to update something. `Etc/UTC` is a valid timezone.

Reply:
I do agree that Etc/UTC is a valid time zone, I don't think adding it explicitly is a good idea. You could change the YANG model PR to make UTC the default as it is in the list already.

Reply:
Done